src/web/client/src/components/Login/Login.test.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { shallow } from "enzyme";
+import { render } from "@testing-library/react";
+import Login from "./Login";
+
+describe("Login component", () => {
+  let loginWrapper: any;
+  let onAuthChangeMock: jest.Mock;
+  let historyMock: object;
+  let eventMock: any;
+  let preventDefaultMock: jest.Mock;
+  beforeEach(() => {
+    // Mock Fetch
+    const mockSuccessResponse = "successfully logged in";
+    const mockTextPromise: Promise<any> = Promise.resolve(mockSuccessResponse);
+    const mockFetchPromise: Promise<any> = Promise.resolve({
+      text: () => mockTextPromise,
+    });
+    jest.spyOn(window, "fetch").mockImplementation(() => mockFetchPromise);
+
+    preventDefaultMock = jest.fn();
+    eventMock = {
+      preventDefault: preventDefaultMock,
+      target: {
+        name: "username",
+        value: "goat",
+      },
```

Comment:
It is so much better to have `onClick= {() => this.handleChange()}` than `onClick=this.handleChange` because of this reason. You don't have to deal with preventDefault and anything else because you're not testing just your code, but you're also testing React's code or browser code.

Reply:
yes